Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -20,140 +19,137 @@ class Credential {
 }
 
 function getRequest(url: string, cred: Credential, strictSSL: boolean): Q.Promise<any> {
-    let defer = Q.defer<any>();
-
-    request
-        .get({url: url, strictSSL: strictSSL}, (err, res, body) => {
-            if (res && body && res.statusCode === 200)  {
-                defer.resolve(JSON.parse(body));
-            } else {
-                if (res && res.statusCode) {
-                   tl.debug(tl.loc('ServerCallErrorCode', res.statusCode)); 
+    const defer = Q.defer<any>();
+
+    request.get({url: url, strictSSL: strictSSL}, (err, res, body) => {
```

Comment:
Why do we have this function at all?  If you used the typed-rest-client it's one line of code and it's awaitable.  No reason to wrap.  typed-rest-client also supports passing along proxy config from agent etc...